doc/api/tls.md

```diff
@@ -1440,6 +1440,11 @@ changes:
     occur in an array. `object.passphrase` is optional. Encrypted keys will be
     decrypted with `object.passphrase` if provided, or `options.passphrase` if
     it is not.
+  * `privateKeyEngine` {string} Name of an OpenSSL engine to get private key
+    from. Should be used together with `engineKey`, otherwise ignored.
+  * `engineKey` {string} Identifier of a private key managed by OpenSSL engine.
+    Should be used together with `privateKeyEngine`, otherwise ignored.
+    Overrides the key set with `key`.
```

Comment:
Would it be a better/cleaner solution to interpret `key` option differently depending on whether `privateKeyEngine` is set (and get rid of `engineKey` alltogether)?

Reply:
I've mixed feelings, that it would integrate better with current APIs is compelling. But, what would this do?

```
key: [
 {pem: '---- BEGIN..'},
  'key-for-sam-in-a-HSM'
],
 keyEngine: '/some/key/hsm.so'
```

Would it do something expected, like choosing the right key for the cert chosen to match the key exchange algorithm?

If not, or if only a HSM key can be used, it might be better to have a different param, like you do now, rather than complicate the docs and meaning of the existing key param.